packages/dataviews/src/dataviews-layouts/utils/get-clickable-item-props.ts

```diff
@@ -1,10 +1,15 @@
-export default function getClickableItemProps< Item >(
-	item: Item,
-	isItemClickable: ( item: Item ) => boolean,
-	onClickItem: ( item: Item ) => void,
-	className: string
-) {
-	if ( ! isItemClickable( item ) ) {
+export default function getClickableItemProps< Item >( {
+	item,
+	isItemClickable,
+	onClickItem,
+	className,
+}: {
+	item: Item;
+	isItemClickable: ( item: Item ) => boolean;
+	onClickItem?: ( item: Item ) => void;
+	className: string;
+} ) {
+	if ( ! isItemClickable( item ) || ! onClickItem ) {
```

Comment:
Here we might want to log a warning when `isItemClickable` returns `true` but there is no `onClickItem` handler. Because that's a nonsense situation.
```
doing_it_wrong( 'DataViews: isItemClickable retured true but there is no onClickItem handler.' );
```

In https://github.com/WordPress/gutenberg/pull/67393#discussion_r1862516356 you say:
> I think there's actually a few more issues where the classnames, role... shouldn't be added if there's no onClickItem. I'll open a follow-up.
but is that really an issue? Adding classnames, role... should be fully determined by `isItemClickable`, that's what it is for.

Reply:
`isItemClickable` is only here if you have some rows that are clickable but not others. and `onClickItem` is whether the whole dataviews support clicking items or not. So I think no, we shouldn't be adding that warning, I think `isItemClickable` doesn't need to be defined and should return true by default.

Reply:
> `isItemClickable` doesn't need to be defined and should return true by default.

The current `defaultIsItemClickable` returns `false` though.

Reply:
> `isItemClickable` doesn't need to be defined and should return true by default.

I see, this patch is also changing the default from `false` to `true`.

Reply:
>  this patch is also changing the default from false to true.

Yes, I actually missed that in the original PR that introduced the onClick handler, I think this is a better default.